src/tool_writeout.c

```diff
@@ -105,14 +107,84 @@ static const struct variable replacements[]={
   {"scheme", VAR_SCHEME},
   {"stdout", VAR_STDOUT},
   {"stderr", VAR_STDERR},
+  {"content", VAR_CONTENT},
+  {"error", VAR_ERROR},
   {NULL, VAR_NONE}
 };
 
+void copyandshift(char **replace, char *escape)
+{
+  int offset;
+  if(replace == NULL || escape == NULL) {
+    return;
+  }
+  offset = strlen(escape);
+  memcpy(*replace, escape, offset);
+  *replace += offset;
+}
+
+char *escape_string(char *output)
+{
+  char *formatStr = NULL, *tmpbuf = output,
+    *replaceStr = NULL, *escapeStr = NULL;
+  int len = 0, offset = 0;
+  if(NULL == output) {
+    return NULL;
+  }
+  len = strlen(output);
+  formatStr = (char *)malloc(2*len + 1);
+  memset(formatStr, 0, 2*len + 1);
+  replaceStr = formatStr;
+  while (*tmpbuf) {
+    switch(*tmpbuf) {
+    case '"':
+      escapeStr = "\\\"";
+      copyandshift(&replaceStr, escapeStr);
```

Comment:
couldn't these invokes then instead just be done like this to save assigning a variable that isn't really used?
~~~c
copyandshift(&replaceStr, "\\\"");